arangod/Graph/ShortestPathOptions.cpp

```diff
@@ -212,18 +215,41 @@ auto ShortestPathOptions::estimateDepth() const noexcept -> uint64_t {
   return 7;
 }
 
+auto ShortestPathOptions::setDefaultWeight(double weight) -> void {
+  if (weight < 0.) {
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_GRAPH_NEGATIVE_EDGE_WEIGHT,
+                                   "negative default weight not allowed");
+  }
+  _defaultWeight = weight;
+}
+
+auto ShortestPathOptions::setWeightAttribute(std::string attribute) -> void {
+  _weightAttribute = attribute;
+}
+
+auto ShortestPathOptions::getDefaultWeight() const -> double {
+  TRI_ASSERT(_defaultWeight > .0);
+  return _defaultWeight;
+}
+
+auto ShortestPathOptions::getWeightAttribute() const -> std::string {
+  return _weightAttribute;
+}
+
 ShortestPathOptions::ShortestPathOptions(ShortestPathOptions const& other,
                                          bool const allowAlreadyBuiltCopy)
     : BaseOptions(other, allowAlreadyBuiltCopy),
       minDepth(other.minDepth),
       maxDepth(other.maxDepth),
       start{other.start},
       end{other.end},
-      weightAttribute{other.weightAttribute},
-      defaultWeight{other.defaultWeight},
       bidirectional{other.bidirectional},
       multiThreaded{other.multiThreaded},
-      _reverseLookupInfos{other._reverseLookupInfos} {}
+      _reverseLookupInfos{other._reverseLookupInfos},
+      _weightAttribute{other._weightAttribute},
+      _defaultWeight{other._defaultWeight} {
+  TRI_ASSERT(other._defaultWeight >= 0.);
```

Comment:
```suggestion
  TRI_ASSERT(other._defaultWeight > 0.);
```

Reply:
TY. will apply in separate commit as I am lazy and want to cherry-pick in a single commit :)

Reply:
reverted